### PR TITLE
Md feature/35/edit a category

### PR DESCRIPTION
**Pull request by Michael-Deitz**
### **Purpose**
To Allow a user with the role admin to create a category

### **Files Changed**
- Created file, and added the class with the needed properties in new `CategoryUpdate.cs`
- Added the endpoint for Editing a Category in `CategoryController.cs`
- Added the fetch for editing a Category in `categoryManager.js`
- Changed file name `CategoryCreate.jsx to `CreateCategoryForm.jsx`
- Added the logic to the handleSubmit so that if categoryId is present then it runs the function that sends the updated category to the controller, otherwise it will run the function that creates a category in `CreateCategoryForm.jsx`
-  added routing for Categories when the user clicks create or edit the user is taken to the corresponding view in `ApplicationView.jsx`

### **To Test**

Fetch ,setup and run according to project README, then confirm the following

- Logged in Admin can click Edit on a category 
- Logged in Admin gets navigated to a form to Edit the category
- Logged in Admin clicks save 
- Logged in Admin gets navigated to the Category Management page and sees the updated category

### CODE REVIEW

File: client/src/components/categories/CreateCategoryForm.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        <h4 className="mt-2" style={{display: 'flex', justifyContent: 'center'}}>Edit a Category</h4>
+        ) : (
+        <h4 className="mt-2" style={{display: 'flex', justifyContent: 'center'}}>Create a Category</h4>
+        )}
+        <Form className="w-50 m-auto"
+            style={{maxWidth:"20rem"}}
+            onSubmit={handleSubmit}
+        >
+            <FormGroup >
+
```
Comment (ccote95) [inline client/src/components/categories/CreateCategoryForm.jsx:50]: Could you remove the white space here on line 50

**Review by ccote95 — APPROVED** [review]
:shipit:

File: client/src/components/categories/CategoryList.jsx
Comment (ztrouy) [inline client/src/components/categories/CategoryList.jsx:1]: Please indent lines 42-62 one more level

**Review by joshuahamrick — APPROVED** [review]

File: client/src/components/categories/CreateCategoryForm.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            else
+            {
+                createCategory(category).then(() => {navigate("/categories")});
+            }
+    };
+
+    return (
+        <>
+        {categoryid ? (
+        <h4 className="mt-2" style={{display: 'flex', justifyContent: 'center'}}>Edit a Category</h4>
```
Comment (ztrouy) [inline client/src/components/categories/CreateCategoryForm.jsx:41]: Indent one level more

File: client/src/components/categories/CreateCategoryForm.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        ) : (
+        <h4 className="mt-2" style={{display: 'flex', justifyContent: 'center'}}>Create a Category</h4>
```
Comment (ztrouy) [inline client/src/components/categories/CreateCategoryForm.jsx:43]: Indent one level more

File: client/src/components/categories/CreateCategoryForm.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        )}
+        <Form className="w-50 m-auto"
+            style={{maxWidth:"20rem"}}
+            onSubmit={handleSubmit}
+        >
+            <FormGroup >               
+                    <Input
+                        type="text"
```
Comment (ztrouy) [inline client/src/components/categories/CreateCategoryForm.jsx:51]: Indent this whole element one level less

File: client/src/components/categories/CreateCategoryForm.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                        value={name}
+                        placeholder="Enter Name For A Category"
+                        onChange={(e) => {
+                        setName(e.target.value);
+                        }}
+                    />
+            </FormGroup>
+                <ButtonToolbar className="gap-2" style={{float: "right"}}>
```
Comment (ztrouy) [inline client/src/components/categories/CreateCategoryForm.jsx:59]: Indent this whole element one level less

File: Controllers/CategoryController.cs [hunk 6]
```diff
@@ -55,6 +55,40 @@ public IActionResult CategoryDelete(int id)
...
+        return Ok(_dbContext.Categories
+        .Where(c => c.Id == id)
+        .Select(c => new CategoryNoNavDTO
+        {
+            Id = c.Id,
+            Name = c.Name
+        }).SingleOrDefault());
+    }
+
+    [HttpPut("edit/{id}")]
```
Comment (ztrouy) [inline Controllers/CategoryController.cs:79]: Remove edit from this path

File: Controllers/CategoryController.cs [hunk 7]
```diff
@@ -55,6 +55,40 @@ public IActionResult CategoryDelete(int id)
...
+    [Authorize()]
```
Comment (ztrouy) [inline Controllers/CategoryController.cs:80]: Missing the roles, or doesn't need the parentheses

**Review by ztrouy — CHANGES_REQUESTED** [review]

**Review by ztrouy — APPROVED** [review]
:shipit:

**Pull request merged**